subsys/bluetooth/host/conn.c

```diff
@@ -1867,6 +1869,13 @@ int bt_conn_get_info(const struct bt_conn *conn, struct bt_conn_info *info)
 		info->le.interval = conn->le.interval;
 		info->le.latency = conn->le.latency;
 		info->le.timeout = conn->le.timeout;
+
+		err = bt_le_get_chan_map(conn);
```

Comment:
Can the chan map change during the lifetime of a connection? If not it'd be good to add a flag to conn->flags so we don't send the HCI command more than once.

Reply:
Yes it can. The master can change it at anytime, to react to bad channels

Reply:
Does it then make sense to store this info at all in bt_conn? Why not just copy it straight to bt_conn_info?

Reply:
Could do that, this is more in line with how read remote version works.

Reply:
Difference is that remote version is static, in fact the controller will cache it also so subsequent requests don't cause any OTA communication.

In general I'd like to understand the use case for this API better. Will this API be called periodically or through some trigger? Would it make sense to have some vendor event for the controller to indicate that the map changed, or is there perhaps even some standard event for that? Meaning, would it make sense to try to pass this info through an async callback rather than on-demand requests?

Reply:
@jhedberg The feature request was only to read the channel map. " It would be useful if there is a function in Zephyr to read the channel map now." I have no idea what it could be used for.

There is no standard HCI event. But i would be careful with adding vendor specific events unless we have really good reason, to have better support for other controllers than the zephyr one.

Reply:
Let's have a conversation with the requestor then to understand this better. I don't think it's wise to go implementing features if we don't have at least a little bit of understanding how they'll be used, when the usage patterns may have substantial impact on what kind of design makes most sense.